### ISSUE TITLE
git segmentation fault

### ISSUE THREAD

**Issue by smcquay** [Homebrew/homebrew-core#23045]
# Please always follow these steps:
- [ x ] Confirmed this is a problem with `brew install`ing one, specific Homebrew/homebrew-core formula (not cask or tap) and not every time you run `brew`? If it's a general `brew` problem please file this issue at https://github.com/Homebrew/brew/issues/new. If it's a `brew cask` problem please file this issue at https://github.com/caskroom/homebrew-cask/issues/new. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [ x ] Ran `brew update` and retried your prior step?
- [ x ] Ran `brew doctor`, fixed all issues and retried your prior step?
- [ x ] Ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ x ] If `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)

trying to clone a repository

- What happened (include command output)

```bash
smcquay@smm:/tmp
 ⌘ rm -rf tcell; /usr/bin/git clone git@github.com:gdamore/tcell.git                                              
Cloning into 'tcell'...
remote: Counting objects: 4168, done.
remote: Compressing objects: 100% (8/8), done.
remote: Total 4168 (delta 3), reused 8 (delta 3), pack-reused 4157
Receiving objects: 100% (4168/4168), 1.68 MiB | 2.21 MiB/s, done.
Resolving deltas: 100% (779/779), done.
smcquay@smm:/tmp
 ⌘ rm -rf tcell; /usr/local/bin/git clone git@github.com:gdamore/tcell.git                                        
Cloning into 'tcell'...
remote: Counting objects: 4168, done.
remote: Compressing objects: 100% (8/8), done.
remote: Total 4168 (delta 3), reused 8 (delta 3), pack-reused 4157
Receiving objects: 100% (4168/4168), 1.68 MiB | 916.00 KiB/s, done.
Resolving deltas: 100% (779/779), done.
zsh: segmentation fault  /usr/local/bin/git clone git@github.com:gdamore/tcell.git
```

- What you expected to happen

When I switch back it works:

```bash
smcquay@smm:/tmp
 ⌘ brew switch git 2.15.1_1                                                                                rc: 1  
Cleaning /usr/local/Cellar/git/2.14.1
Cleaning /usr/local/Cellar/git/2.15.1_1
Cleaning /usr/local/Cellar/git/2.14.2
Cleaning /usr/local/Cellar/git/2.14.3
Cleaning /usr/local/Cellar/git/2.16.0
Cleaning /usr/local/Cellar/git/2.15.1
Cleaning /usr/local/Cellar/git/2.15.0
208 links created for /usr/local/Cellar/git/2.15.1_1
smcquay@smm:/tmp
 ⌘ rm -rf tcell; /usr/local/bin/git clone git@github.com:gdamore/tcell.git                                        
Cloning into 'tcell'...
remote: Counting objects: 4168, done.
remote: Compressing objects: 100% (8/8), done.
remote: Total 4168 (delta 3), reused 8 (delta 3), pack-reused 4157
Receiving objects: 100% (4168/4168), 1.68 MiB | 2.09 MiB/s, done.
Resolving deltas: 100% (779/779), done.
```

- Step-by-step reproduction instructions (by running `brew install` commands)

```bash
$ brew install git
$ /usr/local/bin/git clone git@github.com:gdamore/tcell.git
```

**Comment by ilovezfs** [comment]
That's most likely a `git` issue. Please report it upstream and post the bug report link here.

**Comment by smcquay** [comment]
I reported it and [here's a link](https://public-inbox.org/git/CALwADsGfB10f5+nOFN-pHCt4z1SkWMcvSHn8KokcyCM0V6K-BA@mail.gmail.com/) to the mail thread.

**Comment by ilovezfs** [comment]
Thanks @smcquay!

**Comment by ilovezfs** [comment]
@smcquay upstream fixed this in 2.16.1, which I have just shipped. If you `brew update` and `brew upgrade` the problem should be fixed.

**Comment by smcquay** [comment]
\o/

Tested and works for me. Thanks!
